axlearn/common/launch_trainer_main.py

```diff
@@ -13,6 +13,7 @@ def main(_):
     launch.setup()
     trainer_config = launch_trainer.get_trainer_config()
     trainer_config.set(recorder=config_for_function(lambda: measurement.global_recorder))
+    measurement.start_monitoring()
```

Comment:
I suppose we must do this after `launch.setup` due to `jax.process_index`? If so, maybe we should add a comment to the goodput `start_monitoring` docstring that it assumes jax distributed init.